gsy_e_sdk/setups/Asset_API_scripts/rest_basic_strategies.py

```diff
@@ -0,0 +1,290 @@
+                                        "fee_to_market_maker"
+                                    ]
+                                )
+                            )
+                            * (i / TICKS),
+                            3,
+                        )
+                    )
+                    batt_sell_strategy.append(
+                        round(
+                            market_maker_rate
+                            + self.asset_strategy[area_uuid]["fee_to_market_maker"]
+                            - (
+                                market_maker_rate
+                                + self.asset_strategy[area_uuid]["fee_to_market_maker"]
+                                - med_price
+                            )
+                            * (i / TICKS),
+                            3,
+                        )
+                    )
+
+                self.asset_strategy[area_uuid]["buy_rates"] = batt_buy_strategy
+                self.asset_strategy[area_uuid]["sell_rates"] = batt_sell_strategy
+
+    def post_bid_offer(self, i=0):
+        """Post a bid or an offer to the exchange."""
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            asset_info = area_dict.get("asset_info")
+            if not asset_info:
+                continue
+
+            # Consumption assets
+            required_energy = asset_info.get("energy_requirement_kWh")
+            if required_energy:
+                rate = self.asset_strategy[area_uuid]["buy_rates"][i]
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=rate, energy=required_energy
+                )
+
+            # Generation assets
+            available_energy = asset_info.get("available_energy_kWh")
+            if available_energy:
+                rate = self.asset_strategy[area_uuid]["sell_rates"][i]
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=rate, energy=available_energy
+                )
+
+            # Storage assets
+            buy_energy = asset_info.get("energy_to_buy")
+            if buy_energy:
+                buy_rate = self.asset_strategy[area_uuid]["buy_rates"][i]
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=buy_rate, energy=buy_energy
+                )
+
+            sell_energy = asset_info.get("energy_to_sell")
+            if sell_energy:
+                sell_rate = self.asset_strategy[area_uuid]["sell_rates"][i]
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=sell_rate, energy=sell_energy
+                )
+
+            self.execute_batch_commands()
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def get_assets_name(node: dict) -> dict:
```

Comment:
As this is used also in other scripts, I suggest to put this into a `helper` module and import it in order to reduce code duplication.